src/compositor.rs

```diff
@@ -85,7 +85,7 @@ impl CompositorState {
     {
         let compositor = self.wl_compositor.get()?;
 
-        let surface = compositor.create_surface(qh, data)?;
+        let surface = compositor.create_surface(qh, data);
```

Comment:
Does GlobalError still need an `InvalidId` variant?

Reply:
We can probably remove all of the `InvalidId` variants after this patch, and maybe also some other errors (shm create_buffer can be infallible now).  That might be useful to add in this series.